packages/keyring-controller/src/KeyringController.ts

```diff
@@ -467,7 +477,11 @@ export class KeyringController extends BaseControllerV2<
   async addNewKeyring(
     type: KeyringTypes | string,
     opts?: unknown,
-  ): Promise<Keyring<Json>> {
+  ): Promise<unknown> {
+    if (type === KeyringTypes.qr) {
+      return this.getOrAddQRKeyring();
+    }
```

Comment:
We do this to avoid this issue: https://github.com/MetaMask/core/pull/1702/files#r1334384936